opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -0,0 +1,411 @@
+                let type_id = match val {
+                    Value::StringValue(_) => BondDataType::BT_STRING as u8,
+                    Value::IntValue(_) => BondDataType::BT_INT32 as u8,
+                    Value::DoubleValue(_) => BondDataType::BT_FLOAT as u8, // using float for now
+                    Value::BoolValue(_) => BondDataType::BT_INT32 as u8, // representing bool as int
+                    _ => continue,
+                };
+                fields.push((attr.key.clone(), type_id));
+            }
+        }
+
+        fields
+    }
+
+    /// Calculate schema ID from field specifications
+    fn calculate_schema_id(&self, fields: &[(String, u8)]) -> u64 {
+        use std::collections::hash_map::DefaultHasher;
+        use std::hash::{Hash, Hasher};
+
+        let mut hasher = DefaultHasher::new();
+
+        // Sort fields by name for consistent schema ID
+        let mut sorted_fields = fields.to_vec();
+        sorted_fields.sort_by(|a, b| a.0.cmp(&b.0));
+
+        for (name, type_id) in sorted_fields {
+            name.hash(&mut hasher);
+            type_id.hash(&mut hasher);
+        }
+
+        hasher.finish()
+    }
+
+    /// Get or create schema with field ordering information
+    fn get_or_create_schema(
+        &self,
+        schema_id: u64,
+        field_specs: Vec<(String, u8)>,
+    ) -> (CentralSchemaEntry, Vec<FieldInfo>) {
+        // Check cache first
+        if let Some((schema, field_info)) = self.schema_cache.read().unwrap().get(&schema_id) {
+            let schema_bytes = schema.as_bytes();
+            let schema_md5 = md5::compute(schema_bytes).0;
+
+            return (
+                CentralSchemaEntry {
+                    id: schema_id,
+                    md5: schema_md5,
+                    schema: schema.clone(),
+                },
+                field_info.clone(),
+            );
+        }
+
+        // Create new schema
+        let mut sorted_specs = field_specs.clone();
+        sorted_specs.sort_by(|a, b| a.0.cmp(&b.0));
+
+        let field_defs: Vec<(&str, u8, u16)> = sorted_specs
+            .iter()
+            .enumerate()
+            .map(|(i, (name, type_id))| (name.as_str(), *type_id, (i + 1) as u16))
+            .collect();
+
+        let schema = BondEncodedSchema::from_fields(&field_defs, "OtlpLogRecord", "telemetry"); //TODO - use actual struct name and namespace
+
+        // Create field info for ordered writing
+        let field_info: Vec<FieldInfo> = field_defs
+            .iter()
+            .map(|(name, type_id, order_id)| FieldInfo {
+                name: name.to_string(),
+                type_id: *type_id,
+                order_id: *order_id,
+            })
+            .collect();
+
+        // Cache the schema and field info
+        {
+            let mut cache = self.schema_cache.write().unwrap();
+            cache.insert(schema_id, (schema.clone(), field_info.clone()));
```

Comment:
It looks like the `field_info` data is already contained in the `BondEncodedSchema`'s `DynamicSchema`. Do we need this additional cloning?